src/main/groovy/org/asciidoctor/gradle/AsciidoctorTask.groovy

```diff
@@ -52,6 +53,7 @@ import org.gradle.util.CollectionUtils
  * @author Robert Panzer
  */
 @SuppressWarnings(['MethodCount', 'Instanceof'])
+@CacheableTask
```

Comment:
I have not had time to verify this, but I am not sure whether a class compiled with `@CacheableTask` is backwards-compatible with Gradle < 4.0. If it's not, then this will need to go into a branch. @aalmiray has been talking about a major upgrade to the plugin which will only be GRadle 4.0+ compatible, but that will also means a major version change.

Reply:
Annotations are not binary breaking changes. If they are present in the bytecode but not on the classpath, then they are ignored.
We have a problem with the mapped output directories, though. This is not supported for Gradle < 4.0.